core/commands/generate_repository_partials/kafka_parser.rb

```diff
@@ -0,0 +1,136 @@
+# frozen_string_literal: true
+
+require_relative 'repository_parser_core'
+
+# This module handles the Kafka repository
+module KafkaParser
+  extend RepositoryParserCore
+
+  def self.parse(config, product_version, ui, logger)
+    base_url = config['repo']['path']
+    archive_directories = parse_web_directories(
+      base_url,
+      nil,
+      product_version,
+      ui,
+      logger,
+      extract_field(:base_version, /(\d+.\d+.\d+)/)
+    )
+    releases = archive_directories.each_with_object([]) do |directory, releases|
```

Comment:
Method get_links is low level and is discouraged to be used inside the repository parsers. Why those methods were not appropriate for the task here?

Reply:
Those methods are designated to work with directories only.